app/plugin_api.go

```diff
@@ -34,6 +34,32 @@ func NewPluginAPI(a *App, manifest *model.Manifest) *PluginAPI {
 	}
 }
 
+func (api *PluginAPI) applyDefaultConfigValues(pluginConfig map[string]interface{}) {
+	initialConfig := make(map[string]interface{})
+
+	if api.manifest.SettingsSchema != nil {
+		for _, settings := range api.manifest.SettingsSchema.Settings {
+			initialConfig[strings.ToLower(settings.Key)] = settings.Default
+		}
+	}
+	// Checking if plugin config was specified by user,
+	// if not we just apply default values from the initial config settings and return
+	if len(pluginConfig) == 0 {
+		for key, value := range initialConfig {
+			pluginConfig[key] = value
+		}
+		return
+	}
+	// If plugin config was specified by user we check the values from plugin config and if there is not value for
+	// the key, we will apply default value from the initial plugin settings
+	for configKey, configValue := range initialConfig {
```

Comment:
Should we just range over `api.manifest.SettingsSchema` directly here?

Reply:
@lieut-data Yes, we can iterate directly over the copy of api.manifest.SettingsSchema here, thank you. I will make this changes